exchanges/bybit/bybit_wrapper.go

```diff
@@ -724,79 +591,63 @@ func (by *Bybit) UpdateOrderbook(ctx context.Context, p currency.Pair, assetType
 	if err != nil {
 		return book, err
 	}
-	return orderbook.Get(by.Name, formattedPair, assetType)
+	return orderbook.Get(by.Name, p, assetType)
 }
 
 // UpdateAccountInfo retrieves balances for all enabled currencies
 func (by *Bybit) UpdateAccountInfo(ctx context.Context, assetType asset.Item) (account.Holdings, error) {
 	var info account.Holdings
 	var acc account.SubAccount
+	var accountType string
 	info.Exchange = by.Name
 	switch assetType {
-	case asset.Spot:
-		balances, err := by.GetWalletBalance(ctx)
-		if err != nil {
-			return info, err
-		}
-
-		currencyBalance := make([]account.Balance, len(balances))
-		for i := range balances {
-			currencyBalance[i] = account.Balance{
-				Currency: currency.NewCode(balances[i].CoinName),
-				Total:    balances[i].Total.Float64(),
-				Hold:     balances[i].Locked.Float64(),
-				Free:     balances[i].Total.Float64() - balances[i].Locked.Float64(),
-			}
-		}
-
-		acc.Currencies = currencyBalance
-
-	case asset.CoinMarginedFutures, asset.USDTMarginedFutures, asset.Futures:
-		balances, err := by.GetFutureWalletBalance(ctx, "")
-		if err != nil {
-			return info, err
-		}
-
-		var i int
-		currencyBalance := make([]account.Balance, len(balances))
-		for coinName, data := range balances {
-			currencyBalance[i] = account.Balance{
-				Currency: currency.NewCode(coinName),
-				Total:    data.WalletBalance,
-				Hold:     data.WalletBalance - data.AvailableBalance,
-				Free:     data.AvailableBalance,
+	case asset.Spot, asset.Options,
+		asset.USDCMarginedFutures,
+		asset.USDTMarginedFutures:
+		switch by.AccountType {
```

Comment:
I find there may be some problems with this.
IMO, we should retrieve the account type again here instead of using the previous result.
For example, if a user does not set any credentials in `config.json` on the server side, the `by.AccountType` will be set as 0. 
But if the user specified the new unified account credentials via `gctcli --apikey xxx --apisecret xxx`, which will get the wrong account type (normal).

Maybe `UpdateAccountInfo` should always refresh all the account information, including this `accountType`.

Reply:
If I understand your point correctly, what you are saying is that instead of using a pre-assigned AccountType field, update the fields whenever we call the `UpdateAccountInfo` method.

Reply:
If so, I have moved the `RetrieveAndSetAccountType` method call inside `UpdateAccountInfo`.